js/src/shell/js.cpp

```diff
@@ -5328,44 +5356,84 @@ static bool Parse(JSContext* cx, unsigned argc, Value* vp) {
   ParseInfo parseInfo(cx, allocScope);
 
   RootedScriptSourceObject sourceObject(
-      cx, frontend::CreateScriptSourceObject(cx, options, Nothing()));
+                                        cx, frontend::CreateScriptSourceObject(cx, options, Nothing()));
   if (!sourceObject) {
     return false;
   }
 
-  Parser<FullParseHandler, char16_t> parser(cx, options, chars, length,
-                                            /* foldConstants = */ false,
-                                            parseInfo, nullptr, nullptr,
-                                            sourceObject, goal);
-  if (!parser.checkOptions()) {
-    return false;
-  }
-
-  ParseNode* pn;  // Deallocated once `parser` goes out of scope.
-  if (goal == frontend::ParseGoal::Script) {
-    pn = parser.parse();
-  } else {
-    if (!GlobalObject::ensureModulePrototypesCreated(cx, cx->global())) {
+  if (stableChars.isLatin1()) {
+    const Latin1Char* chars_ = stableChars.latin1Range().begin().get();
+    auto chars = reinterpret_cast<const mozilla::Utf8Unit*>(chars_);
```

Comment:
can the whole then block and else block be moved to a template function with `<typename CharT>` where `CharT` = `Utf8Unit` or `char16_t` ?? so that there's no duplicate for Latin1/twoByte

Reply:
Done, I named it `FullParseTest` and moved it above.